### PR TITLE
[Feature/RPA-675] Contact us form fixes and improvments

### PR DESCRIPTION
**Pull request by FranekMarciniak**
## Description
Changes to "contact us" form from design mocks. Changes to submit message flow.
image
image
image
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-orchestrator-ui/src/src-landing/scss/variables/_palette.scss [hunk 1]
```diff
@@ -11,6 +11,7 @@ $palette-background-primary: #ffffff;
 $palette-background-secondary: #f8f7ff;
 $palette-background-secondary-dark: #eae8fa;
 $palette-background-tertiary: #000000;
+$palette-background-disabled:#F5F5F5 ;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/scss/variables/_palette.scss]: SPACE

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/ContactForm.tsx [hunk 2]
```diff
@@ -59,41 +61,52 @@ const ContactForm: FC = () => {
...
+                onReset={handleReset}
+            >
                 <div className={styles.formRow}>
                     <FormInput
                         name="name"
                         labelValue={translate('Landing.Contact.Form.Name')}
                         type="text"
                         onChange={handleInputChanged}
                         value={formState.name}
+                        disabled={status?.type === FormStatusType.SUCCESS}
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/ContactForm.tsx]: disabled={status?.type === FormStatusType.SUCCESS} is used a lot of times. Try disabling the whole form or if it's not possible create const e.g. isFormSubmitted = status?.type === FormStatusType.SUCCESS

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.module.scss [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@import '#src-landing/scss/main.scss';
+
+.reset {
+    composes: submit from '../FormFields/FormFields.module.scss';
+    padding: $padding-2;
+    justify-self: end;
+    display: flex;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.module.scss:5]: styles order

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+const ResetBox: FC = () => (
+    <button className={styles.reset} type="reset">
+        <Image src={RefreshImg} width={24} height={24} alt=" " />
+    </button>
+);
+
+const AdditionalInfo: FC<{ status: Status }> = ({ status }) => {
+    if (status?.type === FormStatusType.LOADING) {
+        return <Image src={LoaderImg} width={50} height={50} alt=" " />;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx:22]: alt=" " ????

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    if (status?.type === FormStatusType.SUCCESS) {
+        return <ResetBox />;
+    }
+    return <div />;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx:27]: return empty div? imo better use <If condition={} />

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        if (statusType === FormStatusType.SUCCESS) {
+            return 'Landing.Contact.Form.Submit.Sent';
+        }
+        if (statusType === FormStatusType.ERROR) {
+            return 'Landing.Contact.Form.Submit.Error';
+        }
+        if (statusType === FormStatusType.LOADING) {
+            return 'Landing.Contact.Form.Submit.Loading';
+        }
+        return 'Landing.Contact.Form.Submit.Default';
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx:43]: consider `Landing.Contact.Form.Submit.${capitalizeFirstLetter(statusType)}`

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormFields/FormFields.module.scss [hunk 7]
```diff
@@ -12,19 +12,19 @@ $CHECKBOX-SIZE: 26px;
 $CHECKBOX-BORDER-RADIUS: 6px;
 $CHECKBOX-TRANSFORM-Y: -1.2px;
 
-$LABEL-TRANSFORM-Y: calc($font-size-body3 / -2);
+$LABEL-TRANSFORM-Y: -25px;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormFields/FormFields.module.scss:15]: małe jest piękne

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormFields/FormFields.tsx [hunk 8]
```diff
@@ -45,7 +46,8 @@ export const FormCheckbox: FC<CheckboxProps> = (props) => (
     <div className={styles.checkboxWrapper}>
         <input
             className={styles.checkbox}
-            placeholder=" "
+            placeholder={props.placeholder ?? ' '}
+            disabled={props.disabled}
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormFields/FormFields.tsx]: your interface FormInputProps<T> had property disabled as optional. Imo change it there to mandatory or here to optional props?.disabled

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactInformation/ContactInformation.module.scss [hunk 9]
```diff
@@ -29,18 +29,36 @@ $INFO-MAX-WIDTH: 800px;
...
+    .root {
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+        justify-content: center;
+        gap: $padding-6;
+    }
+
+    .summary {
+        gap: $padding-2;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactInformation/ContactInformation.module.scss:59]: right below you overwrite gap with $margin-2

**Review by star-in-the-jar — CHANGES_REQUESTED** [review]
Neat code

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx (hunk 4 above)
Comment (FranekMarciniak) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx:22]: empty alt means its for decoration

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx (hunk 5 above)
Comment (FranekMarciniak) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx:27]: im using grid in this container, so i need an element to keep grid working

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx (hunk 6 above)
Comment (FranekMarciniak) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormButtonGroup/FormButtonGroup.tsx:43]: i think i like it that way, do you think `capitalizeFirstLetter(statusType)` has any benefits?

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormFields/FormFields.module.scss (hunk 7 above)
Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormFields/FormFields.module.scss:15]: seems like a bad idea, there is no other way?

**Review by preludium — DISMISSED** [review]

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormFields/FormFields.tsx [hunk 10]
```diff
@@ -6,14 +6,14 @@ import { CheckboxProps, FormInputProps, SubmitProps } from './FormFields.types';
 
 export const FormInput: FC<FormInputProps<HTMLInputElement>> = (props) => (
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/ContactSection/ContactForm/FormFields/FormFields.tsx]: destructure props {name, disabled, value etc.}

**Review by star-in-the-jar — DISMISSED** [review]

**Review by preludium — APPROVED** [review]

**Pull request merged**